flow/render_status.py

```diff
@@ -0,0 +1,220 @@
+            Context that includes all the information for rendering status output.
+        :type context:
+            dict
+        :param file:
+            Redirect all output to this file, defaults to sys.stdout.
+        :type file:
+            str
+        :param detailed:
+            Print a detailed status of each job.
+        :type detailed:
+            bool
+        :param expand:
+            Present labels and operations in two separate tables.
+        :type expand:
+            bool
+        :param unroll:
+            Separate columns for jobs and the corresponding operations.
+        :type unroll:
+            bool
+        :param compact:
+            Print a compact version of the output.
+        :type compact:
+            bool
+        :param pretty:
+            Prettify the output.
+        :type pretty:
+            bool
+        :param option:
+            Options for rendering format, currently supports:
+            'terminal'(default), 'markdown' and 'html'.
+        :type option:
+            str
+        """
+
+        # use Jinja2 template for status output
+        if template is None:
+            if detailed and expand:
+                template = 'status_expand.jinja'
+            elif detailed and not unroll:
+                template = 'status_stack.jinja'
+            elif detailed and compact:
+                template = 'status_compact.jinja'
+            else:
+                template = 'status.jinja'
+
+        def draw_progressbar(value, total, escape='', width=40):
+            """Visualize progess with a progress bar.
+
+            :param value:
+                The current progress as a fraction of total.
+            :type value:
+                int
+            :param total:
+                The maximum value that 'value' may obtain.
+            :type total:
+                int
+            :param width:
+                The character width of the drawn progress bar.
+            :type width:
+                int
+            """
+
+            assert value >= 0 and total > 0
+            ratio = ' %0.2f%%' % (100 * value / total)
+            n = int(value / total * width)
+            return escape + '|' + ''.join(['#'] * n) + ''.join(['-'] * (width - n)) \
+                          + escape + '|' + ratio
+
+        def job_filter(job_op, scheduler_status_code, all_ops):
+            """filter eligible jobs for status print.
+
+            :param job_ops:
+                Operations information for a job.
+            :type job_ops:
+                OrderedDict
+            :param scheduler_status_code:
+                Dictionary information for status code
+            :type scheduler_status_code:
+                dict
+            :param all_ops:
+                Boolean value indicate if all operations should be displayed
+            :type all_ops:
+                bool
+            """
+
+            if scheduler_status_code[job_op['scheduler_status']] != 'U' or \
```

Comment:
Maybe just use `return conditions` instead of `if conditions: return True; else: return False`